### ISSUE TITLE
PhantomJS crashing while running gulp-uncss task

### ISSUE THREAD

**Issue by Ajeetlakhani** [giakki/uncss#223]
PhantomJS is crashing everytime i run gulp-uncss task. How to optimise uncss to reduce memory usage?(I am running it on VM  with 3.5GB and 2 Core)

**Comment by RyanZim** [comment]
What OS? I'm running vanilla uncss on a Lubuntu VM with ~750MB & 1 core.

**Comment by mikelambert** [comment]
How many CSS files are you passing through your gulp-uncss task? It may unfortunately try to run them in parallel and blow through your memory usage. FWIW, using the postcss version of uncss should fix this (if you are willing to use postcss), since I serialize execution in that pipeline.

(I never felt comfortable doing that on raw uncss and slowing things down, but perhaps we should. Check "serializedQueue" in uncss.js for where this is done.)

**Comment by RyanZim** [comment]
@Ajeetlakhani Is this still an issue for you?

**Comment by RyanZim** [comment]
@mikelambert Should this be closed?

**Comment by mikelambert** [comment]
I guess technically we should be supporting parallel execution requests of phantomjs by the same method I implemented it for postcss plugin (by serializing it). I was scared to do drastic changes at the time since I was new to the project, but given the support load and lack of other maintainer improvements, probably should be willing to make this trade-off...you interested? ;)

**Comment by RyanZim** [comment]
Don't know when I'll get time, collaborating at mde/ejs keeps me quite busy. Perhaps add the `enhancement` label and rename to something more descriptive of the task to be done (or open a new issue)?

If I get time, I'll let everyone know I'm working on it.

**Comment by RyanZim** [comment]
@mikelambert Would serializing PhantomJS requests also fix #136?

**Comment by mikelambert** [comment]
Grunt has a workaround fix, but gulp does not. Fixing this issue would also fix #136 and allow grunt-uncss to remove their hack.

I have https://github.com/giakki/uncss/pull/249 which I believe should fix this, and passes all tests, but I haven't tested it in a proper parallelized environment. @Ajeetlakhani , are you interested in testing my pull request to see if it fixes your issue?
